sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/BoundAttribute.scala

```diff
@@ -62,12 +62,7 @@ case class BoundReference(ordinal: Int, dataType: DataType, nullable: Boolean)
     val javaType = ctx.javaType(dataType)
     val value = ctx.getValue(ctx.INPUT_ROW, dataType, ordinal.toString)
     if (ctx.currentVars != null && ctx.currentVars(ordinal) != null) {
-      val oev = ctx.currentVars(ordinal)
-      ev.isNull = oev.isNull
-      ev.value = oev.value
-      val code = oev.code
-      oev.code = ""
-      ev.copy(code = code)
+      ctx.currentVars(ordinal)
```

Comment:
This is not logically equivalent, the previous one mutate the `ctx.currentVars`, but this one doesn't. cc @davies